### PR TITLE
add auth and endpoint GET and POST for character

### PR DESCRIPTION
**Pull request by phsilveira**

### CODE REVIEW

**Comment by if-ai** [comment]
Awesome This would be extremely useful +Bump

File: extensions/api/blocking_api.py [hunk 1]
```diff
@@ -39,12 +66,23 @@ def do_GET(self):
...
         content_length = int(self.headers['Content-Length'])
-        body = json.loads(self.rfile.read(content_length).decode('utf-8'))
+        try:
+            body = json.loads(self.rfile.read(content_length).decode('utf-8'))
+        except Exception as e:
+            print(e)
+            self.send_error(400)
+            return
+
+        if shared.args.auth_api:
```
Comment (oobabooga) [inline extensions/api/blocking_api.py:84]: Should this be at the top of the `do_POST` function?

File: extensions/api/blocking_api.py [hunk 2]
```diff
@@ -71,6 +109,57 @@ def do_POST(self):
...
 
+        elif self.path == '/api/v1/character':
+            self.send_response(201)
+            self.send_header('Content-Type', 'application/json')
+            self.end_headers()
+
+            name = body['name']
+            context = body['context']
+            greeting = body['greeting']
+            example_dialogue = body['example_dialogue']
```
Comment (oobabooga) [inline extensions/api/blocking_api.py:120]: The example_dialogue field is no longer used in the web UI. It is expected to be part of the context string in character yamls.

File: extensions/api/blocking_api.py [hunk 3]
```diff
@@ -71,6 +109,57 @@ def do_POST(self):
...
+            self.send_header('Content-Type', 'application/json')
+            self.end_headers()
+
+            if body is None:
+                self.simple_json_results("body is None")
+                return
+
+            characters = get_available_characters()
+            for character in characters:
+                delete_character(character)
```
Comment (oobabooga) [inline extensions/api/blocking_api.py:144]: Why does this API endpoint delete all characters?

File: extensions/api/blocking_api.py (hunk 1 above)
Comment (Rifanbyte) [reply extensions/api/blocking_api.py:84]: Yyyy

Comment (Rifanbyte) [reply extensions/api/blocking_api.py:84]: Ok

**Review by Rifanbyte — COMMENTED** [review]
#3731

**Review by Rifanbyte — COMMENTED** [review]
> tujuk @if-ai #3741

**Pull request closed without merging**